playbooks/install_stack.yaml

```diff
@@ -40,35 +40,58 @@
         - /usr/share/openstack-tripleo-heat-templates/environments/ssl/enable-tls.yaml
         - /usr/share/openstack-tripleo-heat-templates/environments/ssl/inject-trust-anchor.yaml
 
-  - name: Generate SSL self-signed certificate
+  - name: Generate SSL self-signed certificate on localhost
     when: ssl_enabled
-    include_role:
-      name: simpleca
-    vars:
-      cert_user: standalone
-      ca_dir: "{{ ansible_env.HOME }}/ssl/ca"
-      cert_dir: "{{ ansible_env.HOME }}/ssl"
-      cert_name: standalone
+    become: false
+    # We run this block on localhost because we don't want to put the CA key on the remote
+    # server, which could lead to security problems.
+    delegate_to: localhost
+    block:
+    - name: Create temporary directory for SSL files
+      tempfile:
+        state: directory
+        suffix: dev-install-ssl
+      register: ssl_dir
+    - name: Generate SSL self-signed certificate
+      when: ssl_enabled
+      include_role:
+        name: simpleca
+      vars:
+        cert_user: standalone
+        ca_dir: "{{ ssl_dir.path }}/ssl/ca"
+        cert_dir: "{{ ssl_dir.path }}/ssl"
+        cert_name: standalone
 
   - name: Prepare the host for SSL
     when: ssl_enabled
     no_log: true
     block:
-      - name: Read SSL certificate
-        slurp:
-          src: "{{ ansible_env.HOME }}/ssl/standalone.crt"
-        register: ssl_cert_output
-        when: ssl_cert is not defined
-      - name: Read SSL key
-        slurp:
-          src: "{{ ansible_env.HOME }}/ssl/standalone.key"
-        register: ssl_key_output
-        when: ssl_ssl_key is not defined
-      - name: Read CA certificate
-        slurp:
-          src: "{{ ansible_env.HOME }}/ssl/ca/simpleca.crt"
-        register: ssl_ca_cert_output
-        when: ssl_ca_cert is not defined
+      - name: Read and clean SSL files
+        become: false
+        delegate_to: localhost
+        block:
+          - name: Read SSL certificate
+            slurp:
+              src: "{{ ssl_dir.path }}/ssl/standalone.crt"
+            register: ssl_cert_output
+            when: ssl_cert is not defined
+          - name: Read SSL key
+            slurp:
+              src: "{{ ssl_dir.path }}/ssl/standalone.key"
+            register: ssl_key_output
+            when: ssl_ssl_key is not defined
```

Comment:
Should be `ssl_key`, I'll address the issue in a separate patch.

Reply:
I meant to create a PR and accidentally pushed to main, sorry :angel: 

https://github.com/shiftstack/dev-install/commit/6724a1daa3b7834871642781913a66672b92d6b7

Reply:
Ah, that will also be why there's a merge conflict.